go/iot/protocol/examples/arduino/protocol.ino

```diff
@@ -75,11 +81,27 @@ void request(void (*func)()) {
     //Serial.println(info);
     switch (code){
       case FAIL:
-        delay(1000);
+        if (failtime == FAILTHRESHOLD){
+          // When there are too many fail attempts, turn on the light to notify
+          // Meanwhile stop increasing the fail attempts counter
+          digitalWrite(LED_BUILTIN, HIGH);
+        }
+        else{
+          failtime = failtime + 1;
+        }
+        delay(random(1 << failtime));
```

Comment:
I don't think you need `random` here. IIUC exponential backoff should be deterministic.

Reply:
But according to the https://en.wikipedia.org/wiki/Exponential_backoff, in the protocol of Ethernet, the exponential time should be random from a range of time slots.

Reply:
I see, then consider adding the random number as a decimal. For example https://cloud.google.com/storage/docs/exponential-backoff#example_algorithm.

Reply:
Okay.